File: DSA-Arrays-Strings/Python/longest_common_prefix.py
def longestCommonPrefix(strs):
    if not strs: return ""
    strs.sort()
    first, last = strs[0], strs[-1]
    i = 0
    while i < len(first) and first[i] == last[i]: i += 1
    return first[:i]
print(longestCommonPrefix(["flower","flow","flight"]))

File: DSA-Arrays-Strings/C++/longest_common_prefix.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;
string longestCommonPrefix(vector<string>& strs) {
    if(strs.empty()) return "";
    sort(strs.begin(), strs.end());
    string a = strs[0], b = strs.back();
    int i = 0;
    while(i < a.size() && a[i] == b[i]) i++;
    return a.substr(0, i);
}
int main() {
    vector<string> strs = {"flower","flow","flight"};
    cout << longestCommonPrefix(strs);
}